src/randgraphs.jl

```diff
@@ -107,9 +107,13 @@ function _suitable(edges::Set{Edge}, potential_edges::Dict{Int, Int})
     return false
 end
 
-function _try_creation(n::Int, k::Int)
+function _try_creation(n::Int,@compat k::Union{Int,Array{Int}})
```

Comment:
We could, but it seems to me that `k::Int` is just a specialized version of `k::Vector{Int}`, no? If that's the case, then you could keep the main function as `k::Vector{Int}` and just call it for `k::Int` after creating the right vector based on `k`.